py/htm/encoders/eye.py

```diff
@@ -0,0 +1,627 @@
+      this is part of the Motor-Control, which is not in the scope of this encoder. 
+
+    Attribute position     (X, Y) coords of eye center within image
+      (wrt [0,0] corner of the image). Default starting position is the center
+      of the image. 
+    Attribute orientation  ... units are radians, the rotation of the sensor
+      (wrt the image)
+    Attribute scale        ... The scale controls the distance between the eye
+      and the image (scale is similar to distance on Z-axis). 
+      Note: In experiments you typically need to manually tune the `scale` for the dataset, 
+      to find a reasonable value through trial and error.
+      More explanation: "When you look at a really big image, you need to take
+      a few steps back to be able to see the whole thing. The scale parameter 
+      allows the eye to do just that, walk forwards and backwards from the image,
+      to find good place to view it from."
+    """
+
+
+    def __init__(self,
+        output_diameter   = 200, # output SDR size is diameter^2
+        sparsityParvo     = 0.2,
+        sparsityMagno     = 0.025,
+        color             = True,):
+        """
+        Argument output_diameter is size of output ... output is a 
+            field of view (image) with circular shape. Default 200. 
+            `parvo/magno_sdr` size is `output_diameter^2`
+        Argument `sparsityParvo` - sparsity of parvo-cellular pathway of the eye.
+            As a simplification, "parvo" cells (P-cells) represent colors, static 
+            object's properties (shape,...) and are used for image classification. 
+            For biologically accurate details see eg. 
+            https://foundationsofvision.stanford.edu/chapter-5-the-retinal-representation/
+            Note: biologically, the ratio between P/M-cells is about 8(P):1(M):(1 rest), see
+            https://www.pnas.org/content/94/11/5900
+        Argument `sparsityMagno` - sparsity of the magno-cellular (M-cells) pathway, which
+            transfers (higly) temporal information in the visual data, as use "used" for
+            motion detection and motion tracking, video processing.
+            For details see @param `sparsityParvo`.
+            TODO: output of M-cells should be processed on a fast TM.
+        Argument color: use color vision (requires P-cells > 0), default true.
+        """
+        self.output_diameter   = output_diameter
+        # Argument resolution_factor is used to expand the sensor array so that
+        # the fovea has adequate resolution.  After log-polar transform image
+        # is reduced by this factor back to the output_diameter.
+        self.resolution_factor = 3
+        self.retina_diameter   = int(self.resolution_factor * output_diameter)
+        # Argument fovea_scale  ... proportion of the image (ROI) which will be covered (seen) by
+        # high-res fovea (parvo pathway)
+        self.fovea_scale       = 0.177
```

Comment:
I have previously misinterpreted this and self.scale. Rename and change this to `fovea_diameter` to be clearer?